src/SingleSignatureDirtyHack.sol

```diff
@@ -269,4 +334,27 @@ abstract contract SingleSignatureDirtyHack is IERC5267, AbstractContext {
         transferParams.token.safeTransferFrom(transferParams.from, transferParams.to, transferParams.requestedAmount);
         return true;
     }
+
+    function transferFromPermit(
+        string calldata typeSuffix,
+        TransferParams calldata transferParams,
+        PackedSignature calldata sig
+    ) external preFlightChecklist(transferParams) returns (bool) {
+        _checkNonce(address(transferParams.token), transferParams.from, transferParams.nonce);
+        bytes32 structHash = _hashStruct(typeSuffix, transferParams);
+        bytes32 signingHash = _encodePermitHash(
+            transferParams.token, transferParams.from, transferParams.nonce, transferParams.amount, structHash
+        );
```

Comment:
you might want to consider actually making the `permit` call as in this contract https://github.com/0xProject/permitandcall/blob/0977908b58cd82ade39527389399221248821572/src/lib/SafePermit.sol#L91-L127

but if you do, be sure not to bubble any revert, because if you do it creates a griefing opportunity.

Reply:
Saw it earlier but, doing exactly what is done there breaks the flow described in Notion of doing a multicall with the permit call and the call to BridgeSettler. Given your suggestion we will do the permit call ourselves? Please confirm before I change it. It seems definitely fine to do anyway

Reply:
Good point :thinking: 

I think it's preferable if we do it ourselves because then we can avoid the extra cold account access of going through MultiCall as well as the copying/encoding overhead. What do you think?

Reply:
Sounds good to me, doing the change